packages/smart-contracts/src/contracts/ERC20TransferableReceivable.sol

```diff
@@ -183,60 +184,10 @@ contract ERC20TransferableReceivable is ERC721, ERC721Enumerable, ERC721URIStora
     receivableInfoMapping[currentReceivableTokenId] = ReceivableInfo({
       tokenAddress: erc20Addr,
       amount: amount,
-      balance: 0
+      balance: 0,
+      requestID: requestID
     });
 
     _mint(owner, currentReceivableTokenId);
-    _setTokenURI(currentReceivableTokenId, newTokenURI);
-  }
-
-  /**
-   * @notice Get an array of all receivable token IDs owned by a specific address.
-   * @param _owner The address that owns the receivable tokens.
-   * @return An array of all receivable token IDs owned by the specified address.
-   */
-  function getTokenIds(address _owner) public view returns (uint256[] memory) {
-    uint256[] memory _tokensOfOwner = new uint256[](ERC721.balanceOf(_owner));
-    uint256 i;
-
-    for (i = 0; i < ERC721.balanceOf(_owner); i++) {
-      _tokensOfOwner[i] = ERC721Enumerable.tokenOfOwnerByIndex(_owner, i);
-    }
-    return (_tokensOfOwner);
-  }
-
-  // The following functions are overrides required by Solidity.
-  /// @dev Overrides ERC721's _beforeTokenTransfer method to include functionality from ERC721Enumerable.
-  function _beforeTokenTransfer(
-    address from,
-    address to,
-    uint256 tokenId
-  ) internal override(ERC721, ERC721Enumerable) {
-    super._beforeTokenTransfer(from, to, tokenId);
-  }
-
-  /// @dev Overrides ERC721's _burn method to include functionality from ERC721URIStorage.
-  function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
-    super._burn(tokenId);
-  }
-
-  /// @dev Overrides ERC721's tokenURI method to include functionality from ERC721URIStorage.
-  function tokenURI(uint256 tokenId)
```

Comment:
Since it's just for aesthetic purpose I think we could totally build a simple solution that makes these NFTs display nicer on marketplaces. A URL that took accepted IDs e.g. request.network/receivableImages/{tokenId} and redirected to a .svg file would be simple to build!